### PR TITLE
Feat/websocketloops

### PR DESCRIPTION
**Pull request by SirTLB**
This PR adds websocket features to the bots. Also, it refactors loop creation by defining reusable components instead of duplicating code in each loop.

- [x] I have read [Migaloo's contribution guidelines](https://github.com/White-Whale-Defi-Platform/white-whale-bots/blob/main/docs/CONTRIBUTING.md).
- [x] My pull request has a sound title and description (not something vague like `Update index.md`)
- [x] All existing and new tests are passing (if any).
- [x] I updated/added relevant documentation.
- [x] The code is formatted properly using eslint

### CODE REVIEW

File: src/core/types/base/mempool.ts [hunk 1]
```diff
@@ -68,6 +68,9 @@ export function decodeMempool(
 					if (ignoreAddresses[msgSend.fromAddress]) {
 						discardIgnored(msgSend.fromAddress, msgSend.toAddress, ignoreAddresses, iteration);
 					}
+					// msgExecuteContract = MsgExecuteContract.fromPartial({
```
Comment (nick134-bit) [inline src/core/types/base/mempool.ts]: remove comment ?

File: src/chains/defaults/messages/getFlashArbMessages.ts [hunk 2]
```diff
@@ -69,7 +69,7 @@ function getWasmMessages(pool: Pool, _offerAsset: RichAsset) {
 	if (isNativeAsset(offerAssetChain.info)) {
 		msg = <DefaultSwapMessage>{
 			swap: {
-				max_spread: String(spread),
+				// max_spread: String(spread),
```
Comment (nick134-bit) [inline src/chains/defaults/messages/getFlashArbMessages.ts:72]: comment needed as reminder ?

File: src/chains/defaults/messages/getFlashArbMessages.ts [hunk 3]
```diff
@@ -78,7 +78,7 @@ function getWasmMessages(pool: Pool, _offerAsset: RichAsset) {
 							: { native: offerAssetChain.info.native_token.denom },
 				},
 
-				belief_price: beliefPriceChain,
+				// belief_price: beliefPriceChain,
```
Comment (nick134-bit) [inline src/chains/defaults/messages/getFlashArbMessages.ts:81]: comment needed as reminder ?

File: src/chains/defaults/messages/getFlashArbMessages.ts (hunk 2 above)
Comment (SirTLB) [reply src/chains/defaults/messages/getFlashArbMessages.ts:72]: yeah wanna keep this one

File: src/chains/defaults/messages/getFlashArbMessages.ts (hunk 3 above)
Comment (SirTLB) [reply src/chains/defaults/messages/getFlashArbMessages.ts:81]: yeah wanna keep this one

File: src/core/types/base/mempool.ts (hunk 1 above)
Comment (SirTLB) [reply src/core/types/base/mempool.ts]: done

**Review by nick134-bit — APPROVED** [review]

**Pull request merged**